src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -51,28 +69,32 @@ def publish_offers_bids(self, message):
         # IDs of markets (Areas) the client is interested in
         filtered_areas_uuids = filters.get("markets")
         market_offers_bids_list_mapping = {}
-        for area_uuid, markets in self.area_uuid_markets_mapping.items():
+        for area_uuid, area_data in self.area_uuid_markets_mapping.items():
             if filtered_areas_uuids and area_uuid not in filtered_areas_uuids:
                 # Client is uninterested in this Area -> skip
                 continue
-            for market in markets:
+            for market in area_data["markets"]:
                 # Cache the market (needed while matching)
                 self.markets_mapping[market.id] = market
```

Comment:
I know its been there from the old PR but are you clearing this cache after some time or after every market cycle?

Reply:
I am not deleting them and I am not sure about the best place to do this, what do you think about using a lru cache here with maxsize?

Reply:
I have tested clearing the cache in the memory cycle and it seems to be working, adapted
Thanks!